File: java/examples/src/main/java/org/pactman/nonprofitcheckplus/examples/cases/Quickstart.java
package org.pactman.nonprofitcheckplus.examples.cases;

import org.pactman.nonprofitcheckplus.Sources;
import org.pactman.nonprofitcheckplus.devtools.FixtureEins;
import org.pactman.nonprofitcheckplus.examples.support.Example;
import org.pactman.nonprofitcheckplus.examples.support.ExampleContext;
import org.pactman.nonprofitcheckplus.models.BmfSource;
import org.pactman.nonprofitcheckplus.models.Nonprofit;
import org.pactman.nonprofitcheckplus.models.OfacSource;
import org.pactman.nonprofitcheckplus.models.Pub78Source;
import org.pactman.nonprofitcheckplus.models.SingleCheckResult;

/**
 * Minimal single nonprofit check.
 *
 * <p>The key is read from the environment. Never hard-code it, and never run
 * this inside an artifact that ships to a user — the key is a private
 * server-side credential.
 */
public final class Quickstart implements Example {

    @Override
    public String id() {
        return "quickstart";
    }

    @Override
    public String title() {
        return "The smallest useful lookup";
    }

    @Override
    public void run(String[] args) {
        String ein = args.length > 0 ? args[0] : FixtureEins.PUBLIC_CHARITY;

        try (ExampleContext context = ExampleContext.withFixtures()) {
            SingleCheckResult result = context.client().nonprofits().check(ein);

            if (result.getNonprofit() == null) {
                System.out.println("No record for EIN " + ein + ".");
                return;
            }

            Nonprofit nonprofit = result.getNonprofit();

            System.out.println("Organization : " + nonprofit.getOrganizationName());
            System.out.println("EIN          : " + nonprofit.getEin());
            System.out.println("Location     : " + nonprofit.getCity() + ", " + nonprofit.getState());
            System.out.println("Profile      : " + nonprofit.getPactmanOrgUrl());
            System.out.println("Checks used  : " + result.getCheckCount());

            // Three source-specific findings, read straight from the response.
            Pub78Source pub78 = Sources.pub78(nonprofit);
            BmfSource bmf = Sources.bmf(nonprofit);
            OfacSource ofac = Sources.ofac(nonprofit);

            System.out.println();
            System.out.println("IRS Publication 78");
            System.out.println(pub78 == null
                    ? "  not returned"
                    : "  listed: " + pub78.getVerified() + ", as of " + pub78.getMostRecent());

            System.out.println();
            System.out.println("IRS Business Master File");
            System.out.println(bmf == null
                    ? "  not returned"
                    : "  status: " + bmf.getStatus()
                            + ", subsection: " + bmf.getSubsectionDescription());

            System.out.println();
            System.out.println("OFAC");
            System.out.println(ofac == null ? "  not returned" : "  " + ofac.getStatus());
        }

        // A syntactically valid EIN and a clean set of findings are not an
        // eligibility decision. Apply your own grantmaking, compliance and risk
        // policy.
    }
}

File: ruby/examples/quickstart.rb
# frozen_string_literal: true

# Minimal single nonprofit check.
#
# Run:  PACTMAN_API_KEY=... bundle exec ruby examples/quickstart.rb [EIN]
#
# The key is read from the environment. Never hard-code it — the key is a private
# server-side credential.

require "pactman/nonprofit_check_plus"

api_key = ENV.fetch("PACTMAN_API_KEY", "")

if api_key.empty?
  warn "Set PACTMAN_API_KEY before running this example."
  exit 1
end

client = Pactman::NonprofitCheckPlus::Client.new(
  api_key: api_key,
  # Production is the default. PACTMAN_BASE_URL is only for a local mock server.
  base_url: ENV.fetch("PACTMAN_BASE_URL", nil),
  timeout: 15
)

ein = ARGV.fetch(0, "41-1787097")
result = client.nonprofits.check(ein)
nonprofit = result.nonprofit

if nonprofit.nil?
  puts "No record for EIN #{ein}."
  exit 0
end

puts "Organization : #{nonprofit.organization_name}"
puts "EIN          : #{nonprofit.ein}"
puts "Location     : #{nonprofit.city}, #{nonprofit.state}"
puts "Profile      : #{nonprofit.pactman_org_url}"
puts "Checks used  : #{result.check_count}"

# Source-specific findings, read straight from the API response.
pub78 = Pactman::NonprofitCheckPlus::Sources.pub78(nonprofit)
bmf = Pactman::NonprofitCheckPlus::Sources.bmf(nonprofit)
ofac = Pactman::NonprofitCheckPlus::Sources.ofac(nonprofit)

puts "\nIRS Publication 78"
puts pub78.nil? ? "  not returned" : "  listed: #{pub78.verified}, as of #{pub78.most_recent}"

puts "\nIRS Business Master File"
puts bmf.nil? ? "  not returned" : "  status: #{bmf.status}, subsection: #{bmf.subsection_description}"

puts "\nOFAC"
puts ofac.nil? ? "  not returned" : "  #{ofac.status}"

# A syntactically valid EIN and a clean set of findings are not an eligibility
# decision. Apply your own grantmaking, compliance and risk policy.
